components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```
@@ -333,30 +463,132 @@ private Response processRequest(final Map<String, String> requestAttributes) {
                         .updateWithPUT(id, resourceString, userManager, attributes, excludedAttributes);
+            } else if (PATCH.class.getSimpleName().equals(httpVerb) && isGroupPermissionsRequest(requestAttributes)) {
+                try {
+                    groupName = getGroupName(id, userManager, groupResourceManager, excludedAttributes);
+
+                    // Decode the resource string and get the permissions to add or remove.
+                    HashMap<String, List<String>> permissionMap = decodePatchOperation(resourceString);
+                    userManager.updatePermissionListOfGroup(groupName, permissionMap.get(SCIMProviderConstants.ADD)
+                            .toArray(new String[0]), permissionMap.get(SCIMProviderConstants.REMOVE)
+                            .toArray(new String[0]));
+
+                    outputPermissions = new JSONArray(Arrays.asList(userManager.getGroupPermissions(groupName)));
+                    scimResponse = new SCIMResponse(ResponseCodeConstants.CODE_OK, outputPermissions
+                            .toString(), responseHeaders);
+                } catch (JSONException e) {
+                    return SupportUtils.buildResponse(groupResourceManager.get(id, userManager, attributes,
+                            excludedAttributes));
+                }
             } else if (PATCH.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager
                         .updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (DELETE.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.delete(id, userManager);
             }
-            return SupportUtils.buildResponse(scimResponse);
+            return SupportUtils.buildResponse(Objects.requireNonNull(scimResponse));
+        } catch (BadRequestException e) {
+            logger.error("The Patch request is invalid. Unable to decode.");
```

e is missing. Always print e. otherwise the actual error will be hidden.